plugin/pkg/shoot/validator/admission.go

```diff
@@ -1193,7 +1193,7 @@ top:
 }
 
 func validateVolumeTypes(constraints []garden.VolumeType, volume, oldVolume *garden.Volume, regions []garden.Region, region string, zones []string) (bool, []string) {
-	if (volume == nil && oldVolume == nil) || volume.Type == nil || (volume != nil && oldVolume != nil && volume.Type != nil && oldVolume.Type != nil && *volume.Type == *oldVolume.Type) {
+	if (volume == nil && oldVolume == nil) || (volume == nil && oldVolume != nil) || volume.Type == nil || (volume != nil && oldVolume != nil && volume.Type != nil && oldVolume.Type != nil && *volume.Type == *oldVolume.Type) {
```

Comment:
The check:
`(volume == nil && oldVolume == nil) || (volume == nil && oldVolume != nil)`
 is the same as:
`volume == nil`
So perhaps we could just use the second.

Reply:
Correct. I changed this accordingly.